### PR TITLE
Update to only add image import types when enabled

### LINKED ISSUE #26170: images.disableStaticImages: if true, the type of `'*.png'` should be string
### What version of Next.js are you using?

11.0.0

### What version of Node.js are you using?

16.3.0

### What browser are you using?

chrome 91

### What operating system are you using?

macos 11.4

### How are you deploying your application?

next start

### Describe the Bug

```tsx
  import staticPng from '../public/static.png'

  <link rel="preload" href={staticPng} as="image" />
  // error: Type 'StaticImageData' cannot be assigned to type 'string'
  <img src={staticPng} alt="" />
  // error: Type 'StaticImageData' cannot be assigned to type 'string'
```

### Expected Behavior

No Error 

Whether it can be add a new global.disable.xxx.d.ts, declare
```ts
declare module '*.png' {
  const content: string
  export default content
}
```
Other media are the same

In this way, when we set `images.disableStaticImages=true` in next.config.js, we can manually modify the next-env.d.ts 
from
`/// <reference types="next/types/global" />`
to
`/// <reference types="next/types/global.disable.xxx" />`

Of course, I have copied the content of next/types/global.d.ts now. The above modification can solve the problem

Now it's just a suggestion 😌

### To Reproduce

use typescript

### PR DESCRIPTION
**Pull request by ijjk**
This updates to only add the static image import types when the feature is enabled to prevent conflicting with existing image handling set-ups. 

Closes: https://github.com/vercel/next.js/issues/26170
x-ref: https://github.com/vercel/next.js/pull/26281

## Bug

- [x] Related issues linked using `fixes #number`
- [x] Integration tests added

## Feature

- [ ] Implements an existing feature request or RFC. Make sure the feature request has been accepted for implementation before opening a PR.
- [ ] Related issues linked using `fixes #number`
- [ ] Integration tests added
- [ ] Documentation added
- [ ] Telemetry added. In case of a feature if it's used or not.

## Documentation / Examples

- [ ] Make sure the linting passes

### CODE REVIEW

File: test/integration/image-component/typescript/test/index.test.js [hunk 1]
```diff
@@ -27,6 +29,21 @@ describe('TypeScript Image Component', () => {
...
+    })
+
+    it('should remove global image types when disabled', async () => {
+      const content = await fs.readFile(nextConfig, 'utf8')
+      await fs.writeFile(
+        nextConfig,
+        content.replace('// disableStaticImages', 'disableStaticImages')
+      )
+      const { code } = await nextBuild(appDir, [], { stderr: true })
+      expect(code).toBe(1)
```
Comment (styfle) [inline test/integration/image-component/typescript/test/index.test.js:43]: How about adding an assertion for stderr here?

**Comment by ijjk** [comment]
# Stats from current PR

<details>
<summary><strong>Default Server Mode</strong> (Decrease detected ✓)</summary>

<br/>

<details>
<summary><strong>General</strong> Overall decrease ✓</summary>

|  | vercel/next.js canary  | ijjk/next.js update/image-type-handling | Change |
| - | - | - | - |
| buildDuration | 11.6s | 11.8s | ⚠️ +241ms |
| buildDurationCached | 2.9s | 2.8s | -49ms |
| nodeModulesSize | 49.2 MB | 49.2 MB | -401 B |

</details>

<details>
<summary><strong>Page Load Tests</strong> Overall increase ✓</summary>

|  | vercel/next.js canary  | ijjk/next.js update/image-type-handling | Change |
| - | - | - | - |
| / failed reqs | 0 | 0 | ✓ |
| / total time (seconds) | 2.179 | 2.214 | ⚠️ +0.04 |
| / avg req/sec | 1147.06 | 1129.13 | ⚠️ -17.93 |
| /error-in-render failed reqs | 0 | 0 | ✓ |
| /error-in-render total time (seconds) | 1.417 | 1.403 | -0.01 |
| /error-in-render avg req/sec | 1763.95 | 1782.04 | +18.09 |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | vercel/next.js canary  | ijjk/next.js update/image-type-handling | Change |
| - | - | - | - |
| framework-HASH.js gzip | 42 kB | 42 kB | ✓ |
| main-HASH.js gzip | 20.2 kB | 20.2 kB | ✓ |
| webpack-HASH.js gzip | 810 B | 810 B | ✓ |
| Overall change | 63 kB | 63 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | ijjk/next.js update/image-type-handling | Change |
| - | - | - | - |
| polyfills-HASH.js gzip | 31.1 kB | 31.1 kB | ✓ |
| Overall change | 31.1 kB | 31.1 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | vercel/next.js canary  | ijjk/next.js update/image-type-handling | Change |
| - | - | - | - |
| _app-HASH.js gzip | 801 B | 801 B | ✓ |
| _error-HASH.js gzip | 3.17 kB | 3.17 kB | ✓ |
| amp-HASH.js gzip | 527 B | 527 B | ✓ |
| css-HASH.js gzip | 330 B | 330 B | ✓ |
| hooks-HASH.js gzip | 903 B | 903 B | ✓ |
| index-HASH.js gzip | 263 B | 263 B | ✓ |
| link-HASH.js gzip | 1.65 kB | 1.65 kB | ✓ |
| routerDirect..HASH.js gzip | 322 B | 322 B | ✓ |
| withRouter-HASH.js gzip | 320 B | 320 B | ✓ |
| bb14e60e810b..30f.css gzip | 125 B | 125 B | ✓ |
| Overall change | 8.41 kB | 8.41 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | ijjk/next.js update/image-type-handling | Change |
| - | - | - | - |
| _buildManifest.js gzip | 392 B | 392 B | ✓ |
| Overall change | 392 B | 392 B | ✓ |

</details>

<details>
<summary><strong>Rendered Page Sizes</strong></summary>

|  | vercel/next.js canary  | ijjk/next.js update/image-type-handling | Change |
| - | - | - | - |
| index.html gzip | 522 B | 522 B | ✓ |
| link.html gzip | 533 B | 533 B | ✓ |
| withRouter.html gzip | 517 B | 517 B | ✓ |
| Overall change | 1.57 kB | 1.57 kB | ✓ |

</details>

</details>
<hr/>
<details>
<summary><strong>Serverless Mode</strong> (Decrease detected ✓)</summary>

<br/>

<details>
<summary><strong>General</strong> Overall decrease ✓</summary>

|  | vercel/next.js canary  | ijjk/next.js update/image-type-handling | Change |
| - | - | - | - |
| buildDuration | 13.2s | 13.1s | -86ms |
| buildDurationCached | 3.8s | 3.7s | -91ms |
| nodeModulesSize | 49.2 MB | 49.2 MB | -401 B |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | vercel/next.js canary  | ijjk/next.js update/image-type-handling | Change |
| - | - | - | - |
| framework-HASH.js gzip | 42 kB | 42 kB | ✓ |
| main-HASH.js gzip | 20.2 kB | 20.2 kB | ✓ |
| webpack-HASH.js gzip | 810 B | 810 B | ✓ |
| Overall change | 63 kB | 63 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | ijjk/next.js update/image-type-handling | Change |
| - | - | - | - |
| polyfills-HASH.js gzip | 31.1 kB | 31.1 kB | ✓ |
| Overall change | 31.1 kB | 31.1 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | vercel/next.js canary  | ijjk/next.js update/image-type-handling | Change |
| - | - | - | - |
| _app-HASH.js gzip | 801 B | 801 B | ✓ |
| _error-HASH.js gzip | 3.17 kB | 3.17 kB | ✓ |
| amp-HASH.js gzip | 527 B | 527 B | ✓ |
| css-HASH.js gzip | 330 B | 330 B | ✓ |
| hooks-HASH.js gzip | 903 B | 903 B | ✓ |
| index-HASH.js gzip | 263 B | 263 B | ✓ |
| link-HASH.js gzip | 1.65 kB | 1.65 kB | ✓ |
| routerDirect..HASH.js gzip | 322 B | 322 B | ✓ |
| withRouter-HASH.js gzip | 320 B | 320 B | ✓ |
| bb14e60e810b..30f.css gzip | 125 B | 125 B | ✓ |
| Overall change | 8.41 kB | 8.41 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | ijjk/next.js update/image-type-handling | Change |
| - | - | - | - |
| _buildManifest.js gzip | 392 B | 392 B | ✓ |
| Overall change | 392 B | 392 B | ✓ |

</details>

<details>
<summary><strong>Serverless bundles</strong> Overall decrease ✓</summary>

|  | vercel/next.js canary  | ijjk/next.js update/image-type-handling | Change |
| - | - | - | - |
| _error.js | 16.9 kB | 16.9 kB | ✓ |
| 404.html | 1.98 kB | 1.98 kB | ✓ |
| 500.html | 1.96 kB | 1.96 kB | ✓ |
| amp.amp.html | 10.8 kB | 10.8 kB | ✓ |
| amp.html | 1.17 kB | 1.17 kB | ✓ |
| css.html | 1.35 kB | 1.35 kB | ✓ |
| hooks.html | 1.23 kB | 1.23 kB | ✓ |
| index.js | 17.2 kB | 17.2 kB | ✓ |
| link.js | 17.5 kB | 17.5 kB | -2 B |
| routerDirect.js | 17.3 kB | 17.3 kB | ✓ |
| withRouter.js | 17.3 kB | 17.3 kB | -2 B |
| Overall change | 105 kB | 105 kB | -4 B |

</details>

</details>
<hr/>
<details>
<summary><strong>Webpack 4 Mode</strong> (Increase detected ⚠️)</summary>

<br/>

<details>
<summary><strong>General</strong> Overall decrease ✓</summary>

|  | vercel/next.js canary  | ijjk/next.js update/image-type-handling | Change |
| - | - | - | - |
| buildDuration | 10.2s | 10.3s | ⚠️ +79ms |
| buildDurationCached | 4.3s | 4.1s | -154ms |
| nodeModulesSize | 49.2 MB | 49.2 MB | -401 B |

</details>

<details>
<summary><strong>Page Load Tests</strong> Overall increase ✓</summary>

|  | vercel/next.js canary  | ijjk/next.js update/image-type-handling | Change |
| - | - | - | - |
| / failed reqs | 0 | 0 | ✓ |
| / total time (seconds) | 2.192 | 2.187 | -0.01 |
| / avg req/sec | 1140.52 | 1143.28 | +2.76 |
| /error-in-render failed reqs | 0 | 0 | ✓ |
| /error-in-render total time (seconds) | 1.434 | 1.328 | -0.11 |
| /error-in-render avg req/sec | 1742.95 | 1881.98 | +139.03 |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | vercel/next.js canary  | ijjk/next.js update/image-type-handling | Change |
| - | - | - | - |
| 677f882d2ed8..HASH.js gzip | 13.3 kB | 13.3 kB | ✓ |
| framework.HASH.js gzip | 41.8 kB | 41.8 kB | ✓ |
| main-HASH.js gzip | 7.99 kB | 7.99 kB | ✓ |
| webpack-HASH.js gzip | 757 B | 757 B | ✓ |
| Overall change | 63.8 kB | 63.8 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | ijjk/next.js update/image-type-handling | Change |
| - | - | - | - |
| polyfills-HASH.js gzip | 31.3 kB | 31.3 kB | ✓ |
| Overall change | 31.3 kB | 31.3 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | vercel/next.js canary  | ijjk/next.js update/image-type-handling | Change |
| - | - | - | - |
| _app-HASH.js gzip | 1.07 kB | 1.07 kB | ✓ |
| _error-HASH.js gzip | 3.84 kB | 3.84 kB | ✓ |
| amp-HASH.js gzip | 536 B | 536 B | ✓ |
| css-HASH.js gzip | 333 B | 333 B | ✓ |
| hooks-HASH.js gzip | 910 B | 910 B | ✓ |
| index-HASH.js gzip | 227 B | 227 B | ✓ |
| link-HASH.js gzip | 1.64 kB | 1.64 kB | ✓ |
| routerDirect..HASH.js gzip | 295 B | 295 B | ✓ |
| withRouter-HASH.js gzip | 292 B | 292 B | ✓ |
| e025d2764813..52f.css gzip | 125 B | 125 B | ✓ |
| Overall change | 9.28 kB | 9.28 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | ijjk/next.js update/image-type-handling | Change |
| - | - | - | - |
| _buildManifest.js gzip | 420 B | 420 B | ✓ |
| Overall change | 420 B | 420 B | ✓ |

</details>

<details>
<summary><strong>Rendered Page Sizes</strong></summary>

|  | vercel/next.js canary  | ijjk/next.js update/image-type-handling | Change |
| - | - | - | - |
| index.html gzip | 568 B | 568 B | ✓ |
| link.html gzip | 581 B | 581 B | ✓ |
| withRouter.html gzip | 560 B | 560 B | ✓ |
| Overall change | 1.71 kB | 1.71 kB | ✓ |

</details>

</details>
Commit: f7cc14d4021f57b833ef5cc44ad685519c2e387e

File: test/integration/image-component/typescript/test/index.test.js (hunk 1 above)
Comment (ijjk) [reply test/integration/image-component/typescript/test/index.test.js:45]: Added a similar check from the above the test

**Review by styfle — APPROVED** [review]
Great work, thanks! 🎉

**Comment by ijjk** [comment]
# Stats from current PR

<details>
<summary><strong>Default Server Mode</strong> (Decrease detected ✓)</summary>

<br/>

<details>
<summary><strong>General</strong> Overall decrease ✓</summary>

|  | vercel/next.js canary  | ijjk/next.js update/image-type-handling | Change |
| - | - | - | - |
| buildDuration | 14.4s | 14.6s | ⚠️ +196ms |
| buildDurationCached | 3.4s | 3.4s | ⚠️ +12ms |
| nodeModulesSize | 49.2 MB | 49.2 MB | -401 B |

</details>

<details>
<summary><strong>Page Load Tests</strong> Overall decrease ⚠️</summary>

|  | vercel/next.js canary  | ijjk/next.js update/image-type-handling | Change |
| - | - | - | - |
| / failed reqs | 0 | 0 | ✓ |
| / total time (seconds) | 2.495 | 2.477 | -0.02 |
| / avg req/sec | 1002.12 | 1009.17 | +7.05 |
| /error-in-render failed reqs | 0 | 0 | ✓ |
| /error-in-render total time (seconds) | 1.394 | 1.418 | ⚠️ +0.02 |
| /error-in-render avg req/sec | 1793.09 | 1763.14 | ⚠️ -29.95 |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | vercel/next.js canary  | ijjk/next.js update/image-type-handling | Change |
| - | - | - | - |
| framework-HASH.js gzip | 42 kB | 42 kB | ✓ |
| main-HASH.js gzip | 20.2 kB | 20.2 kB | ✓ |
| webpack-HASH.js gzip | 810 B | 810 B | ✓ |
| Overall change | 63 kB | 63 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | ijjk/next.js update/image-type-handling | Change |
| - | - | - | - |
| polyfills-HASH.js gzip | 31.1 kB | 31.1 kB | ✓ |
| Overall change | 31.1 kB | 31.1 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | vercel/next.js canary  | ijjk/next.js update/image-type-handling | Change |
| - | - | - | - |
| _app-HASH.js gzip | 801 B | 801 B | ✓ |
| _error-HASH.js gzip | 3.17 kB | 3.17 kB | ✓ |
| amp-HASH.js gzip | 527 B | 527 B | ✓ |
| css-HASH.js gzip | 330 B | 330 B | ✓ |
| hooks-HASH.js gzip | 903 B | 903 B | ✓ |
| index-HASH.js gzip | 263 B | 263 B | ✓ |
| link-HASH.js gzip | 1.65 kB | 1.65 kB | ✓ |
| routerDirect..HASH.js gzip | 322 B | 322 B | ✓ |
| withRouter-HASH.js gzip | 320 B | 320 B | ✓ |
| bb14e60e810b..30f.css gzip | 125 B | 125 B | ✓ |
| Overall change | 8.41 kB | 8.41 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | ijjk/next.js update/image-type-handling | Change |
| - | - | - | - |
| _buildManifest.js gzip | 392 B | 392 B | ✓ |
| Overall change | 392 B | 392 B | ✓ |

</details>

<details>
<summary><strong>Rendered Page Sizes</strong></summary>

|  | vercel/next.js canary  | ijjk/next.js update/image-type-handling | Change |
| - | - | - | - |
| index.html gzip | 522 B | 522 B | ✓ |
| link.html gzip | 533 B | 533 B | ✓ |
| withRouter.html gzip | 517 B | 517 B | ✓ |
| Overall change | 1.57 kB | 1.57 kB | ✓ |

</details>

</details>
<hr/>
<details>
<summary><strong>Serverless Mode</strong> (Decrease detected ✓)</summary>

<br/>

<details>
<summary><strong>General</strong> Overall decrease ✓</summary>

|  | vercel/next.js canary  | ijjk/next.js update/image-type-handling | Change |
| - | - | - | - |
| buildDuration | 16.4s | 16.7s | ⚠️ +246ms |
| buildDurationCached | 4.7s | 4.7s | -24ms |
| nodeModulesSize | 49.2 MB | 49.2 MB | -401 B |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | vercel/next.js canary  | ijjk/next.js update/image-type-handling | Change |
| - | - | - | - |
| framework-HASH.js gzip | 42 kB | 42 kB | ✓ |
| main-HASH.js gzip | 20.2 kB | 20.2 kB | ✓ |
| webpack-HASH.js gzip | 810 B | 810 B | ✓ |
| Overall change | 63 kB | 63 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | ijjk/next.js update/image-type-handling | Change |
| - | - | - | - |
| polyfills-HASH.js gzip | 31.1 kB | 31.1 kB | ✓ |
| Overall change | 31.1 kB | 31.1 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | vercel/next.js canary  | ijjk/next.js update/image-type-handling | Change |
| - | - | - | - |
| _app-HASH.js gzip | 801 B | 801 B | ✓ |
| _error-HASH.js gzip | 3.17 kB | 3.17 kB | ✓ |
| amp-HASH.js gzip | 527 B | 527 B | ✓ |
| css-HASH.js gzip | 330 B | 330 B | ✓ |
| hooks-HASH.js gzip | 903 B | 903 B | ✓ |
| index-HASH.js gzip | 263 B | 263 B | ✓ |
| link-HASH.js gzip | 1.65 kB | 1.65 kB | ✓ |
| routerDirect..HASH.js gzip | 322 B | 322 B | ✓ |
| withRouter-HASH.js gzip | 320 B | 320 B | ✓ |
| bb14e60e810b..30f.css gzip | 125 B | 125 B | ✓ |
| Overall change | 8.41 kB | 8.41 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | ijjk/next.js update/image-type-handling | Change |
| - | - | - | - |
| _buildManifest.js gzip | 392 B | 392 B | ✓ |
| Overall change | 392 B | 392 B | ✓ |

</details>

<details>
<summary><strong>Serverless bundles</strong></summary>

|  | vercel/next.js canary  | ijjk/next.js update/image-type-handling | Change |
| - | - | - | - |
| _error.js | 16.9 kB | 16.9 kB | ✓ |
| 404.html | 1.98 kB | 1.98 kB | ✓ |
| 500.html | 1.96 kB | 1.96 kB | ✓ |
| amp.amp.html | 10.8 kB | 10.8 kB | ✓ |
| amp.html | 1.17 kB | 1.17 kB | ✓ |
| css.html | 1.35 kB | 1.35 kB | ✓ |
| hooks.html | 1.23 kB | 1.23 kB | ✓ |
| index.js | 17.2 kB | 17.2 kB | ✓ |
| link.js | 17.5 kB | 17.5 kB | ✓ |
| routerDirect.js | 17.3 kB | 17.3 kB | ✓ |
| withRouter.js | 17.3 kB | 17.3 kB | ✓ |
| Overall change | 105 kB | 105 kB | ✓ |

</details>

</details>
<hr/>
<details>
<summary><strong>Webpack 4 Mode</strong> (Increase detected ⚠️)</summary>

<br/>

<details>
<summary><strong>General</strong> Overall decrease ✓</summary>

|  | vercel/next.js canary  | ijjk/next.js update/image-type-handling | Change |
| - | - | - | - |
| buildDuration | 12.7s | 13s | ⚠️ +319ms |
| buildDurationCached | 5.1s | 5s | -66ms |
| nodeModulesSize | 49.2 MB | 49.2 MB | -401 B |

</details>

<details>
<summary><strong>Page Load Tests</strong> Overall increase ✓</summary>

|  | vercel/next.js canary  | ijjk/next.js update/image-type-handling | Change |
| - | - | - | - |
| / failed reqs | 0 | 0 | ✓ |
| / total time (seconds) | 2.479 | 2.556 | ⚠️ +0.08 |
| / avg req/sec | 1008.54 | 978.25 | ⚠️ -30.29 |
| /error-in-render failed reqs | 0 | 0 | ✓ |
| /error-in-render total time (seconds) | 1.488 | 1.453 | -0.03 |
| /error-in-render avg req/sec | 1679.76 | 1720.17 | +40.41 |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | vercel/next.js canary  | ijjk/next.js update/image-type-handling | Change |
| - | - | - | - |
| 677f882d2ed8..HASH.js gzip | 13.3 kB | 13.3 kB | ✓ |
| framework.HASH.js gzip | 41.8 kB | 41.8 kB | ✓ |
| main-HASH.js gzip | 7.99 kB | 7.99 kB | ✓ |
| webpack-HASH.js gzip | 757 B | 757 B | ✓ |
| Overall change | 63.8 kB | 63.8 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | ijjk/next.js update/image-type-handling | Change |
| - | - | - | - |
| polyfills-HASH.js gzip | 31.3 kB | 31.3 kB | ✓ |
| Overall change | 31.3 kB | 31.3 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | vercel/next.js canary  | ijjk/next.js update/image-type-handling | Change |
| - | - | - | - |
| _app-HASH.js gzip | 1.07 kB | 1.07 kB | ✓ |
| _error-HASH.js gzip | 3.84 kB | 3.84 kB | ✓ |
| amp-HASH.js gzip | 536 B | 536 B | ✓ |
| css-HASH.js gzip | 333 B | 333 B | ✓ |
| hooks-HASH.js gzip | 910 B | 910 B | ✓ |
| index-HASH.js gzip | 227 B | 227 B | ✓ |
| link-HASH.js gzip | 1.64 kB | 1.64 kB | ✓ |
| routerDirect..HASH.js gzip | 295 B | 295 B | ✓ |
| withRouter-HASH.js gzip | 292 B | 292 B | ✓ |
| e025d2764813..52f.css gzip | 125 B | 125 B | ✓ |
| Overall change | 9.28 kB | 9.28 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | ijjk/next.js update/image-type-handling | Change |
| - | - | - | - |
| _buildManifest.js gzip | 420 B | 420 B | ✓ |
| Overall change | 420 B | 420 B | ✓ |

</details>

<details>
<summary><strong>Rendered Page Sizes</strong></summary>

|  | vercel/next.js canary  | ijjk/next.js update/image-type-handling | Change |
| - | - | - | - |
| index.html gzip | 568 B | 568 B | ✓ |
| link.html gzip | 581 B | 581 B | ✓ |
| withRouter.html gzip | 560 B | 560 B | ✓ |
| Overall change | 1.71 kB | 1.71 kB | ✓ |

</details>

</details>
Commit: cdff814dffa5272d3678fb1355afebd306eb3d62

**Comment by ijjk** [comment]
## Failing test suites 
Commit: cdff814dffa5272d3678fb1355afebd306eb3d62

**test/integration/basic/test/index.test.js** 
- Basic Features > should polyfill Node.js modules

<details>
<summary>Expand output</summary>

  ● Basic Features › should polyfill Node.js modules

    thrown: "Exceeded timeout of 300000 ms for a test.
    Use jest.setTimeout(newTimeout) to increase the timeout value, if this is a long-running test."

      35 |   afterAll(() => killApp(context.server))
      36 |
    > 37 |   it('should polyfill Node.js modules', async () => {
         |   ^
      38 |     const browser = await webdriver(context.appPort, '/node-browser-polyfills')
      39 |
      40 |     console.error({

      at integration/basic/test/index.test.js:37:3
      at Object.<anonymous> (integration/basic/test/index.test.js:18:1)

</details>

**Comment by kripod** [comment]
Thank you, this is a great improvement over the current situation 😊

Is there a way to omit SVG module type imports e.g. when using a custom webpack config? _(See #26281)_

**Comment by klarstrup** [comment]
Yo this wipes out my customized next-env.d.ts :[

**Comment by styfle** [comment]
Next.js owns the `next-env.d.ts` file but you can customize `tsconfig.json` to add your own declaration file, for example:

```json
{
  "include": ["next-env.d.ts", "my-custom-file.d.ts", "**/*.ts", "**/*.tsx"]
}
```

We'll make sure to document this usage, thanks!

**Comment by pipech** [comment]
How to override `*.svg` modules ?

Adding `*.svg` into `my-custom-file.d.ts` doesn't work and there has to be workaround, is this intended ?

https://duncanleung.com/next-js-typescript-svg-any-module-declaration
https://stackoverflow.com/questions/68103844/how-to-override-next-js-svg-module-declaration

**Comment by brc-dd** [comment]
Hello @pipech, I am the author of that [question](https://stackoverflow.com/q/68103844/11613622) (and the [answer](https://stackoverflow.com/a/68129058/11613622)), and the blog is also probably based on that answer and [this one](https://stackoverflow.com/a/67641345/11613622) itself. Are you having a specific problem with the workaround, or are you simply here to ask the team to handle this in the framework itself? If the latter is the case, then I would like to support that too.

Related Issues: #26357, #26400, #26533, #26560, #26622

**Comment by brc-dd** [comment]
Hello @pipech, I am the author of [that question](https://stackoverflow.com/q/68103844/11613622) (and [the answer](https://stackoverflow.com/a/68129058/11613622)), and [the blog](https://duncanleung.com/next-js-typescript-svg-any-module-declaration) is also probably based on that answer and [this one](https://stackoverflow.com/a/67641345/11613622) itself. Are you having a specific problem with the workaround, or are you simply here to request the team to handle this in the framework itself 'cause it is not quite developer friendly as it used to before?

If the latter is the case, then I would like to support that too.
Related Issues: #26357, #26400, #26533, #26560, #26622

**Comment by smhmd** [comment]
@pipech Another workaround is to somehow not use `default` export. SVGR by default uses `ReactComponent`, so it should work directly. With [`svg-sprite-loader`](https://github.com/JetBrains/svg-sprite-loader/tree/master/examples/custom-runtime-generator), for example, instead of using `export default class` (from Icon.jsx), use `export const ReactComponent = class`.

Then, in your custom `d.ts` file, do this:

```typescript
declare module '*.svg' {
  const ReactComponent: (props: React.SVGProps<SVGSVGElement>) => JSX.Element
  export { ReactComponent }
}
```

Now, instead of `import GlobeSVG from 'assets/globe.svg'`, you import `{ ReactComponent as GlobeSVG }` like CRA makes you do.

Ofc, you can define an `assets/index.ts` file where you:

```js
export { ReactComponent as GlobeSVG } from './globe.svg'

// and

import { GlobeSVG } from 'assets'
```

**Pull request merged**